lib/bundler/plugin/installer.rb

```diff
@@ -38,40 +36,42 @@ def definition.lock(*); end
 
     private
 
-      def install_git(names, version, options)
-        uri = options.delete(:git)
-        options["uri"] = uri
+      def extract_sources_from_options!(options)
+        sources = {}
 
+        sources[:git] = options.delete(:git) if options[:git]
+        sources[:file] = options.delete(:file) if options[:file]
+        sources[:source] = options.delete(:source) if options[:source]
```

Comment:
Do we actually need to remove these from `options`?

Reply:
...looking at how `prepare_source_list` interacts with the `sources` hash you're building up here, I think you're ok populating it with `nil` so ` sources[:git] = options[:git]` would be legit, or even

```
def extract_sources_from_options!(options)
  sources = {}

  [:file, :git, :sources].each do |source|
    sources[source] = options[source]
  end

  sources
end
```

Reply:
> Do we actually need to remove these from options?

So! The rationale behind the removal is to pass whatever is needed downstream, to make it clearer when reading the calling class file (installer.rb) rather than checking the called one (git.rb). In particular, when I was reading the code, this structure would have made it a bit easier.

However I'm fairly neutral with both philosophies (it's still easy to go downstream and see what's used); I'll apply this change :+1:.